src/commands/periscope/helpers/periscopehelper.ts

```diff
@@ -0,0 +1,265 @@
+import * as vscode from 'vscode';
+import * as k8s from 'vscode-kubernetes-tools-api';
+import { getSASKey, LinkDuration } from '../../utils/azurestorage';
+import { parseResource } from '../../../azure-api-utils';
+import * as ast from 'azure-arm-storage';
+import { PeriscopeStorage, PeriscopeHTMLInterface } from '../models/storage';
+import * as amon from 'azure-arm-monitor';
+import * as htmlhandlers from "handlebars";
+import * as path from 'path';
+import * as fs from 'fs';
+import AksClusterTreeItem from '../../../tree/aksClusterTreeItem';
+const tmp = require('tmp');
+
+const {
+    BlobServiceClient,
+    StorageSharedKeyCredential
+} = require("@azure/storage-blob");
+
+export async function getClusterDiagnosticSettings(
+    cluster: AksClusterTreeItem
+): Promise<amon.MonitorManagementModels.DiagnosticSettingsCategoryResourceCollection | undefined> {
+    try {
+        // Get daignostic setting via diagnostic monitor
+        const diagnosticMonitor = new amon.MonitorManagementClient(cluster.root.credentials, cluster.root.subscriptionId);
+        const diagnosticSettings = await diagnosticMonitor.diagnosticSettingsOperations.list(cluster.id!);
+
+        return diagnosticSettings;
+    } catch (e) {
+        vscode.window.showErrorMessage(`Error fetching cluster diagnostic monitor: ${e}`);
+        return undefined;
+    }
+}
+
+export async function chooseStorageAccount(
+    diagnosticSettings: amon.MonitorManagementModels.DiagnosticSettingsResourceCollection | undefined,
+): Promise<string | undefined> {
+    /*
+        Check the diagnostic setting is 1 or more than 1:
+          1. For the scenario of 1 storage account in diagnostic settings - Pick the storageId resource and get SAS.
+          2. For the scenario for more than 1 then show VsCode quickPick to select and get SAS of selected.
+    */
+
+    if (diagnosticSettings && diagnosticSettings.value!.length > 1) {
```

Comment:
Suggest using the early-exit strategy e.g.

```
if (!diagnosticSettings || !diagnosticSettings.value) {
  return undefined;
}

if (diagnosticSettings.value.length === 1) {
  // return the only one
}

// do the picker and return the result of that
```

This avoids the extra indentation and makes your `if` conditions less complex.  You dispose of the 'easy' cases early and then focus on the hard case (which, by the way, you could extract to a helper function to keep the logic here _even more_ focused on the which-case-are-we-in thing).